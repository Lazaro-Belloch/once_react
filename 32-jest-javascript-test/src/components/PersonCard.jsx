import React from 'react'

const PersonCard = (props) => {


    const {persona}  = props;
    //const persona = props.persona;

    return (
        <div className="person-card">
            <div class="ui card">
                <div class="image">
                    <img src="https://semantic-ui.com/images/avatar2/large/kristy.png" />
                </div>
                <div class="content">
                    <a class="header">{persona.name}</a>
                    <div class="meta">
                        <span class="date">Miembro desde el a√±o {persona.joined}</span>
                    </div>
                    <div>
                        Hola
                    </div>
                    <div class="description">
                        {persona.quote}
                    </div>
                </div>
                <div class="extra content">
                    <a>
                        <i class="user icon"></i>
                        {persona.friends} Amigos
                    </a>
                </div>
            </div>
        </div>
    )
}

export default PersonCard
