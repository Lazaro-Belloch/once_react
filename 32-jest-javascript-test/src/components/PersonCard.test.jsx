import React from 'react'
import renderer from 'react-test-renderer'
import PersonCard from './PersonCard'


const personaDePrueba = {
    name: "Juanita",
    joined: 2015,
    quote: "Soy la mejor",
    friends: 10
}


test('se pintan todos los elementos', () => {
    const card = renderer.create(
        <PersonCard persona={personaDePrueba}></PersonCard>
    ).toJSON()
    console.log("card", card); 
    // Primera comprobación
    expect(card.type).toBe("div");
    // Segunda comprobación
    expect(card).toMatchSnapshot()
})




test('dos más dos son cuatro', () => {
    expect(2+2).toBe(4)
})
