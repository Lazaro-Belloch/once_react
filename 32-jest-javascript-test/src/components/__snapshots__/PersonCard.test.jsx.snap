// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`se pintan todos los elementos 1`] = `
<div
  className="person-card"
>
  <div
    class="ui card"
  >
    <div
      class="image"
    >
      <img
        src="https://semantic-ui.com/images/avatar2/large/kristy.png"
      />
    </div>
    <div
      class="content"
    >
      <a
        class="header"
      >
        Juanita
      </a>
      <div
        class="meta"
      >
        <span
          class="date"
        >
          Miembro desde el a√±o 
          2015
        </span>
      </div>
      <div>
        Hola
      </div>
      <div
        class="description"
      >
        Soy la mejor
      </div>
    </div>
    <div
      class="extra content"
    >
      <a>
        <i
          class="user icon"
        />
        10
         Amigos
      </a>
    </div>
  </div>
</div>
`;
